fdbserver/DataDistribution.actor.cpp

```diff
@@ -1559,11 +1620,94 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 		return true;
 	}
 
+	int calculateHealthyServerCount() {
+		int serverCount = 0;
+		for (auto i =server_info.begin(); i != server_info.end(); ++i) {
+			if (!server_status.get(i->first).isUnhealthy()) {
+				++serverCount;
+			}
+		}
+		return serverCount;
+	}
+
+	int calculateHealthyMachineCount() {
+		int totalHealthyMachineCount = 0;
+		for (auto m : machine_info) {
+			if (isMachineHealthy(m.second)) {
+				++totalHealthyMachineCount;
+			}
+		}
+
+		return totalHealthyMachineCount;
+	}
+
+	// compare function for map with key type as Reference<TCMachineTeamInfo>
+	struct TCMachineTeamInfoReferenceCompare
+	{
+	   bool operator() (const Reference<TCMachineTeamInfo>& lhs, const Reference<TCMachineTeamInfo>& rhs) const
+	   {
+		   return lhs->machineIDs < rhs->machineIDs;
+	   }
+	};
+
+
+	// Find the machine team with the least number of process teams
+	 int getMachineTeamWithLeastProcessTeams(Reference<TCMachineTeamInfo> &mt) {
```

Comment:
Add a field in machine_team info to count the number of server teams in the machine team. this filed makes the function faster.